junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/DiscoverySelectorResolver.java

```diff
@@ -64,7 +64,7 @@ public void resolveSelectors(EngineDiscoveryRequest request, TestDescriptor engi
 		request.getSelectorsByType(UniqueIdSelector.class).forEach(selector -> {
 			javaElementsResolver.resolveUniqueId(selector.getUniqueId());
 		});
-		pruneTree(engineDescriptor);
+		engineDescriptor.pruneTree();
```

Comment:
Create https://github.com/junit-team/junit5/issues/732 -- and will leave the prune here alone.